src/matrix/mod.rs

```
@@ -541,7 +541,11 @@ impl<T: Any + Float> Matrix<T> {
             (self[[0, 1]] * self[[1, 0]] * self[[2, 2]]) -
             (self[[0, 2]] * self[[1, 1]] * self[[2, 0]])
         } else {
-            let (l, u, p) = self.lup_decomp().expect("Could not compute LUP decomposition.");
+            let (l, u, p) = match self.lup_decomp() {
+                Ok(x) => x,
+                Err(Error { kind: ErrorKind::DivByZero, .. }) => return T::zero(),
+                _ => { panic!("Could not compute LUP decomposition."); }
```

This is definitely a welcome change!

I wonder if it is still worth us panicking here (instead of returning a `Result`). I think LUP can only fail in this case and so we should never see the panic. If that assumption is correct then I think keeping the panic is appropriate.